underscore.js

```diff
@@ -533,12 +533,20 @@
   // Produce a duplicate-free version of the array. If the array has already
   // been sorted, you have the option of using a faster algorithm.
   // Aliased as `unique`.
-  _.uniq = _.unique = function(array, isSorted, iteratee, context) {
+  _.uniq = _.unique = function(array, isSorted, iteratee, context, predicate) {
```

Comment:
I would rather split this out into another method and make this implementation internal:
1) leave `_.uniq` alone
2) add `function uniqWith(array, predicate, context)`